lib/class-controller.php

```diff
@@ -31,14 +31,105 @@ class Controller {
 	 */
 	private static Controller $instance;
 
+	/**
+	 * Enable aggregations based on WP Categories.
+	 *
+	 * @param array $args Arguments to pass to the adapter's facet configuration.
+	 * @return Controller The instance of the class to allow for chaining.
+	 */
+	public function enable_category_aggregation( array $args = [] ): Controller {
+		if ( isset( $this->adapter ) ) {
+			$this->adapter->enable_category_aggregation();
+			$defaults = [
+				'count'     => 1000,
+				'name'      => 'Categories',
+				'startOpen' => false,
+				'type'      => 'category',
+			];
+
+			$args = wp_parse_args( $args, $defaults );
+			$this->adapter->add_facet_config( $args );
+		}
+		return $this;
+	}
+
+	/**
+	 * Enable faceting on empty search query strings.
+	 *
+	 * @return Controller The instance of the class to allow for chaining.
+	 */
+	public function enable_empty_search_faceting(): Controller {
+		if ( isset( $this->adapter ) ) {
+			$this->adapter->enable_empty_search_faceting();
+		}
+		return $this;
+	}
+
+	/**
+	 * Enable aggregations based on WP Tags.
+	 *
+	 * @param array $args Arguments to pass to the adapter's facet configuration.
+	 * @return Controller The instance of the class to allow for chaining.
+	 */
+	public function enable_tag_aggregation( array $args = [] ): Controller {
+		if ( isset( $this->adapter ) ) {
+			$this->adapter->enable_tag_aggregation();
+			$defaults = [
+				'count'     => 1000,
+				'name'      => 'Tags',
+				'startOpen' => false,
+				'taxonomy'  => 'post_tag',
+				'type'      => 'taxonomy',
+			];
+
+			$args = wp_parse_args( $args, $defaults );
+			$this->adapter->add_facet_config( $args );
+		}
+		return $this;
+	}
+
+	/**
+	 * Enables an aggregation based on post dates.
+	 *
+	 * @param array $args Arguments to pass to the adapter's facet configuration.
+	 * @return Controller The instance of the class to allow for chaining.
+	 */
+	public function enable_post_date_aggregation( array $args = [] ): Controller {
+		if ( isset( $this->adapter ) ) {
+			$this->adapter->enable_post_date_aggregation();
+			$defaults = [
+				'count'             => 1000,
+				'calendar_interval' => 'month',
```

Comment:
Recommend defaulting to year here. I think bucketing by year is likely the most common scenario, and would result in the smallest number of buckets, and folks could zoom in from there if they need/want to.